src/components/CustomerQuota/NoQuota/NoQuotaCard.tsx

```diff
@@ -214,8 +216,18 @@ export const NoQuotaCard: FunctionComponent<NoQuotaCard> = ({
     ? differenceInSeconds(now, latestTransactionTime)
     : -1;
 
+  const appealProductsCategories = allProducts
+    ?.filter((product) => product.categoryType === "APPEAL")
+    .map((product) => product.category);
+
   const hasAppealProduct =
-    allProducts?.some((policy) => policy.categoryType === "APPEAL") ?? false;
+    (appealProductsCategories &&
```

Comment:
I think you should add a few test cases to test the functionality of this. Maybe move this out of NoQuotaCard, and then test it in `NoQuotaCard.test.ts`?

see the function `getLatestTransactionTime` to see what I mean!

applies to appealProductCategories too